stablehlo/tests/interpret_dot_general.mlir

```diff
@@ -0,0 +1,33 @@
+// RUN: stablehlo-translate --interpret -split-input-file %s
+
+func.func @dot_general_op_test_si64() {
```

Comment:
Can we simplify these names from `dot_general_op_test_foo` to just `dot_general_foo`? Or even better, just `foo`? Not sure what the current convention is.

Reply:
Current convention is to have `dot_general_op_test_foo`. I agree this is a long and unnecessary name. Maybe it's better to rename it such that (in order) the test copied from spec should be `spec_example` or `example`, ones testing element types be `si64` or something equivalent, and the rest be some concise name that doesn't include the obvious prefixes. I'll create a ticket for that and keep the name for now as this is the current convention.

Reply:
Issue: #1596